src/main/webapp/js/index.js

```diff
@@ -12,8 +12,94 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/* Date range picker handler:
- * https://www.daterangepicker.com/
+
+/** Fetch receipts from the server and add them to the DOM. */
+async function searchReceipts() {
+  const label = document.getElementById('search-input').value;
+  const response = await fetch(`/search-receipts?label=${label}`);
+  const receipts = await response.json();
+
+  clearExistingReceipts();
+  displayReceipts(label, receipts);
+}
+
+/**
+ * Clear out existing receipts display.
+ *
+ * Note: 3 childNodes make up a receipt card.
+ * The first 3 nodes are the template receipt card and each subsequent
+ * 3 are clones of the template. The template must never be deleted or
+ * it'll be unusable in the future.
+ */
+function clearExistingReceipts() {
+  const receiptsDisplay = document.getElementById('receipts-display');
+  const numChildNodes = receiptsDisplay.childNodes.length;
```

Comment:
Maybe you could use the querySelector to get all the `<div class="col-md-6">` instead of all child nodes to delete each "receipt card" element at a time instead of each node at a time.